### ISSUE TITLE
Fatal Python error: _enter_buffered_busy: could not acquire lock for <_io.BufferedWriter name='<stdout>'> at interpreter shutdown, possibly due to daemon threads Python runtime state: finalizing (tstate=0x556fc0f4b170)

### ISSUE THREAD

**Issue by domadaaaa** [Project-MONAI/research-contributions#297]
how to resolve this error?

**Environment (please complete the following information):**
OS:
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.3 LTS"
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
5.4.0-42-generic

 **Virtual env:**
python==3.9
Package                       Version                     Editable project location
----------------------------- --------------------------- ----------------------------
absl-py                       1.4.0
acvl-utils                    0.2
alembic                       1.11.2
anyio                         3.7.1
attrs                         23.1.0
batchgenerators               0.25
bcrypt                        4.0.1
beautifulsoup4                4.12.2
blinker                       1.6.2
cachetools                    5.3.0
certifi                       2023.7.22
cffi                          1.15.1
charset-normalizer            3.2.0
click                         8.1.6
cloudpickle                   2.2.1
cmake                         3.27.2
connected-components-3d       3.12.2
contourpy                     1.1.0
cryptography                  41.0.3
cucim                         23.6.0
cycler                        0.11.0
databricks-cli                0.17.7
dicom2nifti                   2.4.8
dicomweb-client               0.59.1
diskcache                     5.6.1
docker                        6.1.3
dynamic-network-architectures 0.2
ecdsa                         0.18.0
einops                        0.3.0
entrypoints                   0.4
exceptiongroup                1.1.2
expiring-dict                 1.1.0
expiringdict                  1.2.2
fastapi                       0.95.0
filelock                      3.11.0
fire                          0.5.0
Flask                         2.3.2
fonttools                     4.42.0
fsspec                        2023.6.0
future                        0.18.3
gdown                         4.7.1
girder-client                 3.1.17
gitdb                         4.0.10
GitPython                     3.1.32
google-auth                   2.22.0
google-auth-oauthlib          1.0.0
graphviz                      0.20.1
greenlet                      2.0.2
grpcio                        1.56.2
gunicorn                      20.1.0
h11                           0.14.0
httpcore                      0.16.3
httpx                         0.23.3
huggingface-hub               0.16.4
idna                          3.4
imagecodecs                   2023.8.12
imageio                       2.31.1
importlib-metadata            6.8.0
importlib-resources           6.0.1
itk                           5.3.0
itk-core                      5.3.0
itk-filtering                 5.3.0
itk-io                        5.3.0
itk-numerics                  5.3.0
itk-registration              5.3.0
itk-segmentation              5.3.0
itsdangerous                  2.1.2
Jinja2                        3.1.2
joblib                        1.3.1
jsonschema                    3.2.0
kiwisolver                    1.4.4
lazy_loader                   0.3
linecache2                    1.0.0
lit                           16.0.6
lmdb                          1.4.1
Mako                          1.2.4
Markdown                      3.4.4
MarkupSafe                    2.1.3
matplotlib                    3.7.2
MedPy                         0.4.0
mlflow                        2.5.0
monai                         0.7.0
mpmath                        1.3.0
networkx                      3.1
nibabel                       3.1.1
ninja                         1.11.1
nnunet                        1.7.1
nnunetv2                      2.1.2                       /work/home/pazhou_095/nnUNet
nptyping                      2.5.0
numpy                         1.23.5
numpymaxflow                  0.0.5
nvidia-cublas-cu11            11.10.3.66
nvidia-cuda-cupti-cu11        11.7.101
nvidia-cuda-nvrtc-cu11        11.7.99
nvidia-cuda-runtime-cu11      11.7.99
nvidia-cudnn-cu11             8.5.0.96
nvidia-cufft-cu11             10.9.0.58
nvidia-curand-cu11            10.2.10.91
nvidia-cusolver-cu11          11.4.0.1
nvidia-cusparse-cu11          11.7.4.91
nvidia-nccl-cu11              2.14.3
nvidia-nvtx-cu11              11.7.91
oauthlib                      3.2.2
opencv-python-headless        4.7.0.72
openslide-python              1.1.2
packaging                     23.1
pandas                        2.0.3
passlib                       1.7.4
Pillow                        10.0.0
pip                           23.2.1
protobuf                      3.20.3
psutil                        5.9.5
pyarrow                       12.0.1
pyasn1                        0.5.0
pyasn1-modules                0.3.0
pycparser                     2.21
pydantic                      1.10.12
pydicom                       2.3.1
pydicom-seg                   0.4.1
PyJWT                         2.8.0
pynetdicom                    2.0.2
pynrrd                        1.0.0
pyparsing                     3.0.9
pyrsistent                    0.19.3
PySocks                       1.7.1
python-dateutil               2.8.2
python-dotenv                 1.0.0
python-gdcm                   3.0.22
python-jose                   3.3.0
python-multipart              0.0.6
pytorch-ignite                0.4.11
pytz                          2023.3
PyWavelets                    1.4.1
PyYAML                        6.0
querystring-parser            1.2.4
regex                         2023.6.3
requests                      2.28.2
requests-oauthlib             1.3.1
requests-toolbelt             0.10.1
retrying                      1.3.4
rfc3986                       1.5.0
rsa                           4.9
safetensors                   0.3.2
schedule                      1.1.0
scikit-image                  0.21.0
scikit-learn                  1.3.0
scipy                         1.11.1
seaborn                       0.12.2
setuptools                    68.0.0
shapely                       2.0.1
SimpleITK                     2.2.1
six                           1.16.0
smmap                         5.0.0
sniffio                       1.3.0
sortedcontainers              2.4.0
soupsieve                     2.4.1
SQLAlchemy                    2.0.19
sqlparse                      0.4.4
starlette                     0.26.1
sympy                         1.12
tabulate                      0.9.0
tensorboard                   2.13.0
tensorboard-data-server       0.7.1
tensorboardX                  2.1
termcolor                     2.3.0
threadpoolctl                 3.2.0
tifffile                      2023.7.18
timeloop                      1.0.2
tokenizers                    0.13.3
torch                         1.10.0a0+git2040069.dtk2210
torchaudio                    0.10.0+git9dcc7a1.dtk2210
torchvision                   0.10.0a0+e04d001.dtk2210
tqdm                          4.59.0
traceback2                    1.4.0
transformers                  4.31.0
triton                        2.0.0
typing_extensions             4.7.1
tzdata                        2023.3
unittest2                     1.1.0
urllib3                       1.26.16
uvicorn                       0.21.1
watchdog                      3.0.0
websocket-client              1.6.1
Werkzeug                      2.3.6
wheel                         0.38.4
yacs                          0.1.8
zipp                          3.16.2
